tools/depcheck/pkg/cmd/pin_test.go

```diff
@@ -0,0 +1,282 @@
+package cmd
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/openshift/origin/tools/depcheck/glide"
+)
+
+var yamlFileData = `package: github.com/openshift/origin
+excludeDirs:
+- dir/one
+- dir/two
+import:
+- package: github.com/pkg/one
+  version: v3.2.8
+- package: github.com/pkg/two
+  version: bde6d0fbd9dec5a997e906611fe0364001364c41
+
+# this comment should be preserved
+- package: github.com/pkg/three
+  version: release-1.0
+`
+
+var lockFileData = `hash: 7308ac17a0294aa86998765c0fc8bd42c73a215584e73cffbedb0c38d7b31c1d
+updated: 2017-12-14T09:07:15.649059681-05:00
+imports:
+- name: github.com/pkg/one
+  version: 75cd24fc2f2c2a2088577d12123ddee5f54e0675
+- name: github.com/pkg/two
+  version: 3b1ae45394a234c385be014e9a488f2bb6eef821
+- name: github.com/pkg/three
+  version: fc380db5fc13c6dd71a5b0bf2af0d182865d1b1d
+- name: github.com/pkg/four
+  version: bbbad097214e2918d8543d5201d12bfd7bca254d
+  repo: https://github.com/openshift/pkg-four
+- name: github.com/pkg/five
+  version: 593d64559f7600f29581a3ee42177f5dbded27a9
+`
+
+func setupAndRunPinOpts(out, errout io.Writer) (*PinImportsOpts, error) {
```

Comment:
Looks like you're missing a test that adds yaml and lock imports and sees the final output coming out correctly.